packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -260,13 +262,15 @@ abstract contract CrossDomainMessenger is
         uint256 _value,
         uint256 _minGasLimit,
         bytes calldata _message
-    ) external payable nonReentrant whenNotPaused {
+    )
+        external
+        payable
+        perMessageNonReentrant(
```

Comment:
I think it'd be much simpler if this were just a mapping inside the `CrossDomainMessenger` and didn't require an additional contract to inherit from. Thoughts?

Reply:
Originally pulled it out at the request of @tynes - Removing the contract in favor of a non-assembly version now.